airbyte-cdk/python/unit_tests/sources/file_based/test_scenarios.py

```diff
@@ -178,12 +219,18 @@ def read(capsys, tmp_path, scenario):
             make_file(tmp_path / "catalog.json", scenario.configured_catalog(SyncMode.full_refresh)),
         ],
     )
-    captured = capsys.readouterr()
-    return [
-        msg
-        for msg in (json.loads(line) for line in captured.out.splitlines())
-        if msg["type"] == "RECORD"
-    ]
+    captured = capsys.readouterr().out.splitlines()
+    return (
```

Comment:
nit: this method might be easier to use if it returned a Mapping[AirbyteMessage.Type: List[AirbyteMessage]]